pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -245,3 +384,48 @@ func newIdentity(userAcc *toolchainv1alpha1.UserAccount, user *userv1.User) *use
 func getIdentityName(userAcc *toolchainv1alpha1.UserAccount) string {
 	return fmt.Sprintf("%s:%s", config.GetIdP(), userAcc.Spec.UserID)
 }
+
+// AddOrUpdateStatusCondition adds the new condition to the array of conditions. If there is already a condition
+// with the same type in the current condition array then the condition is updated
+// Also returns a bool flag which indicates if the conditions where updated
+// TODO: move to toolchain-common
+func AddOrUpdateStatusCondition(currentConditions []toolchainv1alpha1.Condition, newCondition toolchainv1alpha1.Condition) ([]toolchainv1alpha1.Condition, bool) {
+	var oldCondition *toolchainv1alpha1.Condition
+	var index int
+	var newConditions []toolchainv1alpha1.Condition
+	if currentConditions == nil {
+		newConditions = make([]toolchainv1alpha1.Condition, 0, 1)
+	} else {
+		newConditions = make([]toolchainv1alpha1.Condition, 0, len(currentConditions))
+		for i, condition := range currentConditions {
+			newConditions[i] = currentConditions[i]
```

Comment:
Yeah, I had something similar to that at first. But then decided that it’s probably better not to modify the current conditions param/slice and always create a new one. Just to avoid potential situations when such modification can be unexpected/undesired.
But maybe it’s not that critical and having a simpler/clearer function implementation is worth it...

Reply:
I see. Well, it's still possible to always return a new array - just instead of setting to the specific index:
```go
currentConditions[i] = newCondition
return currentConditions, true
```
you can do something like this:
```go
return append(append(currentConditions[:i], newCondition), currentConditions[i+1:]...), true
```

Reply:
good point

Reply:
Done in https://github.com/codeready-toolchain/toolchain-common/pull/16